scanner/matcher/updater/vuln/updater_test.go

```diff
@@ -258,3 +265,291 @@ func TestUpdater_Initialized(t *testing.T) {
 		assert.Contains(t, `"level":"error"`, b.String())
 	})
 }
+
+func TestUpdater_Import(t *testing.T) {
+	ctx := zlog.Test(context.Background(), t)
+	ctrl := gomock.NewController(t)
+
+	// Represents one vulnerability or enrichment iteration.
+	type iteration struct {
+		O   *driver.UpdateOperation
+		V   []*claircore.Vulnerability
+		E   []*driver.EnrichmentRecord
+		Err string // Set to fail the iteration.
+	}
+
+	// Mock jsonblob.Iterate with fake iterations. Operations, vulnerabilities and
+	// enrichments are yielded in order. If Err is set, yields the operation followed
+	// by an error, unless the operation is nil, then yields error immediately.
+	mockIterate := func(t *testing.T, ops ...*iteration) {
+		prev := jsonBlobIterateFunc
+		t.Cleanup(func() {
+			jsonBlobIterateFunc = prev
+		})
```

Comment:
to avoid this, perhaps we should make `jsonBlobIterateFunc` a value of the struct instead of a global

Reply:
It's not global. That's a package-level ref. We can't avoid having a ref to the function if we want to mock it in tests. Both package-level and struct-ref are good enough for this purpose.